### PR TITLE
build images weekly, without cache

### PR DESCRIPTION
**Pull request by Emilgardis**

### CODE REVIEW

**Comment by Emilgardis** [comment]
this gives me the idea to also start making progress into buildx, it'd be cool if `build-docker-images` had a flag for outputting a bake file

File: .github/workflows/daily.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build xtask
+        run: cargo build -p xtask
+      - name: Build and push Docker image
+        id: build-docker-image
+        timeout-minutes: 60
```
Comment (Emilgardis) [inline .github/workflows/daily.yml:25]: this will probably not work XD

**Comment by Emilgardis** [comment]
depends on #791

**Comment by Emilgardis** [comment]
this could also maybe implement parts of #610, checking that mounts works correctly

File: .github/workflows/weekly.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build xtask
+        run: cargo build -p xtask
+      - name: Build Docker image
+        id: build-docker-image
+        run: cargo xtask build-docker-image -v --no-cache --from-ci --tag daily
```
Comment (Alexhuszagh) [inline .github/workflows/weekly.yml:25]: Should be `--tag weekly`.

**Review by Alexhuszagh — CHANGES_REQUESTED** [review]
Minor nit.

File: xtask/src/build_docker_image.rs [hunk 3]
```diff
@@ -101,7 +101,17 @@ pub fn build_docker_image(
...
+                    format!(
+                        "{}{}",
+                        m.target,
+                        if let Some(sub) = &m.sub {
+                            format!(".{sub}")
+                        } else {
+                            <_>::default()
+                        }
+                    )
+                })
```
Comment (Emilgardis) [inline xtask/src/build_docker_image.rs:114]: fixed this too, apparently it went away in some rebase somewhere

**Review by Alexhuszagh — APPROVED** [review]
lgtm

**Pull request merged**